src/platform/windows/mod.rs

```diff
@@ -0,0 +1,1479 @@
+
+            let ret = kernel32::CreateIoCompletionPort(*self.handle,
+                                                       iocp,
+                                                       *self.handle as winapi::ULONG_PTR,
+                                                       0);
+            if ret == ptr::null_mut() {
+                return Err(WinError::last("CreateIoCompletionPort"));
+            }
+
+            self.iocp = iocp;
+
+            {
+                let mut reader = self.reader.borrow_mut();
+                // make sure that the reader has a read in flight,
+                // otherwise a later select() will hang
+                try!(reader.ensure_is_reading());
+                reader.iocp = iocp;
+            }
+
+            Ok(())
+        }
+    }
+
+    // Do a pipe connect.  Only used for one-shot servers
+    fn accept(&mut self) -> Result<(),WinError> {
+        unsafe {
+            let mut ov = create_overlapped();
+            reset_overlapped(&mut ov);
+            let ov_ptr = ov.deref() as *const _ as *mut winapi::OVERLAPPED;
+
+            let ok = kernel32::ConnectNamedPipe(*self.handle, ov_ptr);
+
+            // we should always get FALSE with async IO
+            assert!(ok == winapi::FALSE);
+            let err = GetLastError();
+
+            // did we successfully connect? (it's reported as an error [ok==false])
+            if err == winapi::ERROR_PIPE_CONNECTED {
+                dd!("[$ {:?}:{:?}] accept (PIPE_CONNECTED)", self.iocp, *self.handle);
+                return Ok(());
+            }
+
+            // This is a weird one -- if we create a named pipe (like we do
+            // in new(), the client connects, sends data, then drops its handle,
+            // a Connect here will get ERROR_NO_DATA -- but there may be data in
+            // the pipe that we'll be able to read.  So we need to go do some reads
+            // like normal and wait until ReadFile gives us ERROR_NO_DATA.
+            if err == winapi::ERROR_NO_DATA {
+                dd!("[$ {:?}:{:?}] accept (ERROR_NO_DATA)", self.iocp, *self.handle);
+                return Ok(());
+            }
+
+            // was it an actual error?
+            if err != winapi::ERROR_IO_PENDING {
+                dd!("[$ {:?}:{:?}] accept error -> {}", self.iocp, *self.handle, err);
+                return Err(WinError::last("ConnectNamedPipe"));
+            }
+
+            // The connect is pending; wait for it to complete
+            let mut nbytes: u32 = 0;
+            let ok = kernel32::GetOverlappedResult(*self.handle, ov_ptr, &mut nbytes, winapi::TRUE);
+            if ok == winapi::FALSE {
+                return Err(WinError::last("GetOverlappedResult[ConnectNamedPipe]"));
+            }
+
+            Ok(())
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct OsIpcSender {
+    // The client hande itself
+    handle: WinHandle,
+}
+
+unsafe impl Send for OsIpcSender { }
+unsafe impl Sync for OsIpcSender { }
```

Comment:
I guess you have seen that I went ahead with this change in https://github.com/servo/ipc-channel/pull/115 ...